src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowUpdate.cs

```diff
@@ -0,0 +1,110 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.EditData.Contracts;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.EditData.UpdateManagement
+{
+    /// <summary>
+    /// An update to apply to a row of a result set. This will generate an UPDATE statement.
+    /// </summary>
+    public sealed class RowUpdate : RowEditBase
+    {
+        private const string UpdateStatement = "UPDATE {0} SET {1} {2}";
+        private const string UpdateStatementHekaton = "UPDATE {0} WITH (SNAPSHOT) SET {1} {2}";
+
+        private readonly Dictionary<int, CellUpdate> cellUpdates;
+        private readonly IList<DbCellValue> associatedRow;
+
+        /// <summary>
+        /// Constructs a new RowUpdate to be added to the cache.
+        /// </summary>
+        /// <param name="rowId">Internal ID of the row that will be updated with this object</param>
+        /// <param name="associatedResultSet">Result set for the rows of the object to update</param>
+        /// <param name="associatedMetadata">Metadata provider for the object to update</param>
+        public RowUpdate(long rowId, ResultSet associatedResultSet, IEditTableMetadata associatedMetadata)
+            : base(rowId, associatedResultSet, associatedMetadata)
+        {
+            cellUpdates = new Dictionary<int, CellUpdate>();
+            associatedRow = associatedResultSet.GetRow(rowId);
+        }
+
+        /// <summary>
+        /// Constructs an update statement to change the associated row.
+        /// </summary>
+        /// <returns>An UPDATE statement</returns>
+        public override string GetScript()
+        {
+            // Build the "SET" portion of the statement
+            IEnumerable<string> setComponents = cellUpdates.Select(kvp =>
+            {
+                string formattedColumnName = SqlScriptFormatter.FormatIdentifier(kvp.Value.Column.ColumnName);
```

Comment:
>kvp.Value [](start = 81, length = 9)

Please give this a useful name like "updateInfo". It's really hard when hopping into code to track down what the value in a kvp is "meant" to represent.